src/clients/catalog.ts

```diff
@@ -113,6 +114,25 @@ export class Catalog extends JanusClient {
     })
   }
 
+  public search(
+    query: string,
+    authMethod: AuthMethod = 'AUTH_TOKEN',
+    tracingConfig?: RequestTracingConfig
+  ) {
+    const metric = 'catalog-search'
+    const token = getAuthToken(this.context, authMethod)
+
+    return this.http.get(routes.search(query), {
```

Comment:
The `query` will be enough because you can add query parameters to the URL (_e.g:_ `/blusa?map=ft`), but wouldn't be better to type the rightful accepted query params and pass them to `this.http`'s _params_?

Something like [this](https://github.com/vtex-apps/search-resolver/blob/master/node/clients/search.ts#L268) is also fine.

But this should go aligned with your needs, LGTM.